modules/storage-api/src/test/java/org/apache/ignite/internal/storage/index/impl/BinaryTupleComparator.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.index.impl;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.comparingDouble;
+import static java.util.Comparator.comparingInt;
+import static java.util.Comparator.comparingLong;
+import static org.apache.ignite.internal.storage.index.impl.ComparatorUtils.comparingNull;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.stream.IntStream;
+import org.apache.ignite.internal.schema.BinaryTuple;
+import org.apache.ignite.internal.schema.BinaryTupleSchema;
+import org.apache.ignite.internal.schema.NativeType;
+import org.apache.ignite.internal.storage.index.SortedIndexDescriptor;
+import org.apache.ignite.internal.storage.index.SortedIndexDescriptor.ColumnDescriptor;
+
+/**
+ * Comparator implementation for comparting {@link BinaryTuple}s on a per-column basis.
+ */
+class BinaryTupleComparator implements Comparator<ByteBuffer> {
+    /**
+     * Actual comparator implementation.
+     */
+    private final Comparator<ByteBuffer> delegate;
+
+    /** Index descriptor. */
+    private final SortedIndexDescriptor indexDescriptor;
+
+    /**
+     * Creates a comparator for a Sorted Index identified by the given descriptor.
+     */
+    BinaryTupleComparator(SortedIndexDescriptor descriptor, BinaryTupleSchema schema) {
+        this.indexDescriptor = descriptor;
+
+        this.delegate = comparing(bytes -> new BinaryTuple(schema, bytes), binaryTupleComparator());
+    }
+
+    /**
+     * Creates a comparator that compares two {@link BinaryTuple}s by comparing individual columns.
+     */
+    private Comparator<BinaryTuple> binaryTupleComparator() {
+        return IntStream.range(0, indexDescriptor.indexColumns().size())
```

Comment:
I expect comparison to be a very hot method. Let's avoid all these streams and lambdas, ok?

Reply:
It's a test class, why should we bother? I expect the production implementation to be different

Reply:
Ok, again, I didn't realize it was in a "test" folder

Reply:
I removed these streams